To: vim_dev@googlegroups.com
Subject: Patch 7.3.466
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.466
Problem:    Get ml_get error when ":behave mswin" was used and selecting
	    several lines.  (A. Sinan Unur)
Solution:   Adjust the end of the operation. (Christian Brabandt)
Files:	    src/ops.c


*** ../vim-7.3.465/src/ops.c	2012-02-22 17:37:55.000000000 +0100
--- src/ops.c	2012-03-07 19:27:08.000000000 +0100
***************
*** 1957,1962 ****
--- 1957,1965 ----
  	    ++curwin->w_cursor.lnum;
  	    del_lines((long)(oap->line_count - 2), FALSE);
  
+ 	    if (delete_last_line)
+ 		oap->end.lnum = curbuf->b_ml.ml_line_count;
+ 
  	    n = (oap->end.col + 1 - !oap->inclusive);
  	    if (oap->inclusive && delete_last_line
  		    && n > (int)STRLEN(ml_get(oap->end.lnum)))
*** ../vim-7.3.465/src/version.c	2012-03-07 19:16:49.000000000 +0100
--- src/version.c	2012-03-07 19:29:52.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     466,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
153. You find yourself staring at your "inbox" waiting for new e-mail
     to arrive.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
